docs/references/api/aggregate_functions.rst

```diff
@@ -0,0 +1,341 @@
+.. _aggregate_functions:
+
+Aggregate Functions
+###################
+
+Aggregate functions allow you to summarize data by performing calculations across groups of rows. For instance, if you have an ``orders`` table that has an ``amount`` column, you could use an aggregate function to get the sum of the ``amount`` column, either for all rows, or for each group of rows that share specific values, for instance all rows that share the same ``order_date``.
+
+.. note::
+ Aggregate functions are *disabled* by default in PostgREST, as without appropriate safeguards, aggregate functions can create performance problems. See :ref:`db-aggregates-enabled` for further details.
+
+PostgREST supports the following aggregate functions: ``avg()``, ``count()``, ``max()``, ``min()``, and ``sum()``. Please refer to the `section on aggregate functions in the PostgreSQL documentation <https://www.postgresql.org/docs/current/functions-aggregate.html>`_ for a detailed explanation of these functions.
+
+To use an aggregate function, you append the function to a value in the ``select`` parameter, like so:
+
+.. tabs::
+
+  .. code-tab:: http
+
+    GET /orders?select=amount.sum() HTTP/1.1
+
+  .. code-tab:: bash Curl
+
+    curl "http://localhost:3000/orders?select=amount.sum()"
+
+With the above query, PostgREST will return a single row with a single column named ``sum`` that contains the sum of all the values in the ``amount`` column:
+
+.. code-block:: json
+
+  [
+    {
+      "sum": 1234.56
+    }
+  ]
+
+You can use multiple aggregate functions just by adding more columns with aggregate functions to the ``select`` parameter.
+
+To group by other columns, you simply add those columns to the ``select`` parameter. For instance:
+
+.. tabs::
+
+  .. code-tab:: http
+
+    GET /orders?select=amount.sum(),amount.avg(),order_date HTTP/1.1
+
+  .. code-tab:: bash Curl
+
+    curl "http://localhost:3000/orders?select=amount.sum(),amount.avg(),order_date"
+
+This will return a row for each unique value in the ``order_date`` column, with the sum and average of the ``amount`` column for all rows that share the same ``order_date``:
+
+.. code-block:: json
+  [
+    {
+      "sum": 1234.56,
+      "avg": 123.45,
+      "order_date": "2023-01-01"
+    },
+    {
+      "sum": 2345.67,
+      "avg": 234.56,
+      "order_date": "2023-01-02"
+    }
+  ]
+
+.. note::
+  Aggregate functions work alongside other PostgREST features, like :ref:`h_filter`, :ref:`json_columns`, and :ref:`ordering`. Please note at this time aggregate functions are not compatible with :ref:`domain_reps`. Additionally, PostgreSQL's ``HAVING`` clause and ordering by aggregated columns are not yet supported.
+
+The Case of ``count()``
+===========================
+
+.. note::
```

Comment:
@steve-chavez I added this note. not sure if you agree with it, so let me know if you think it should be removed/modified.

Reply:
Fully agree. Though there are lots of `count` users, not something we can break anytime soon. Still it's good to discourage its use.